### PR TITLE
ARM template to deploy a VM installed with tWAS base edition

### PR DESCRIPTION
**Pull request by majguo**
## Description
This PR is to address the following issues:
* resolve WASdev/azure.websphere-traditional.singleserver/issues/1

## Change summary
* Implement an ARM template to install a RHEL 8.4 VM on Azure with tWAS base and IBM Java JDK installed
* Implement cloud-init script to do entitlement check and upgrade
* Add entitlement check in `install.sh` to fail fast if the user account is unentitled

## Test cases
The following cases have already been verified before sending out the PR:
* Successfully deployed a RHEL 8.4 VM on Azure where with tWAS base and IBM Java JDK installed:
   * Installed com.ibm.websphere.BASE.v90_9.0.5010.20211108_1200 to the /datadrive/IBM/WebSphere/Base/V9/ directory.
   * Installed com.ibm.java.jdk.v8_8.0.7000.20211025_1430 to the /datadrive/IBM/WebSphere/Base/V9/ directory. 
* Successfully created a private image based on the above VM;
* Successfully deployed a RHEL 8.4 VM with tWAS base and IBM Java JDK installed using the above private image, for an entitled IBMid;
* Successfully deployed a RHEL 8.4 VM **without** tWAS base and IBM Java JDK installed using the above private image, for a non-entitled IBMid;

Signed-off-by: Jianguo Ma <jiangma@microsoft.com>

### CODE REVIEW

File: twas-base/pom.xml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.ibm.websphere.azure</groupId>
+    <artifactId>azure.websphere-base.image</artifactId>
+    <version>1.0.0</version>
+    
+    <parent>
+        <groupId>com.microsoft.azure.iaas</groupId>
+        <artifactId>azure-javaee-iaas-parent</artifactId>
+        <version>1.0.11</version>
```
Comment (edburns) [inline twas-base/pom.xml:28]: Why not 1.0.13 (currently the latest)?

**Review by edburns — APPROVED** [review]
Please answer my question about the use of the azure-javaee-iaas pom version 1.0.11 instead of 1.0.13?

File: twas-base/pom.xml (hunk 1 above)
Comment (majguo) [reply twas-base/pom.xml:28]: No special reason. Just because it's copied from `twas-nd` sub-directory.

Comment (majguo) [reply twas-base/pom.xml]: Upgraded to `1.0.13` in f13ab08.

**Review by edburns — APPROVED** [review]

File: twas-base/src/main/scripts/virtualimage.properties [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+WAS_BASE_TRADITIONAL=com.ibm.websphere.BASE.v90
+IBM_JAVA_SDK=com.ibm.java.jdk.v8
+REPOSITORY_URL=https://www.ibm.com/software/repositorymanager/entitled
+WAS_ND_VERSION_ENTITLED=ND.v90_9.0.5007
```
Comment (git4rk) [inline twas-base/src/main/scripts/virtualimage.properties:17]: We should check for Base Entitlement:
WAS_BASE_VERSION_ENTITLED=BASE.v90_9.0.5009

Comment (majguo) [reply twas-base/src/main/scripts/virtualimage.properties]: Good catch @git4rk , I've fixed it with commit ca51c84 based on your comment, thanks.

**Review by git4rk — APPROVED** [review]

**Pull request merged**